File: RustSideStory/SampleSource/swift_history.m
// Old and busted

// Not so functional programming
NSMutableArray *filteredTrains = [NSMutableArray array];
for (Train *train in trains) {
    if ([train.trainCategory isEqualToString:@"Commuter"] || 
        [train.trainCategory isEqualToString:@"LongDistance"]) {
        [filteredTrains addObject:train];
    }
}

// Closures
[self doSomethingWithFirstValue:3
                    secondValue:7
                          block:^(int firstValue, int secondValue) {
    return firstValue + secondValue;
}];

// Nested calls
[[myObject theArray] insertObject:[myObject objectToInsert] atIndex:0];

File: RustSideStory/SampleSource/swift_history.swift
// New hotness

// Functional programming!
let filteredTrains = trains?.filter({ $0.trainCategory == "Commuter" || $0.trainCategory == "Long-distance" }) ?? []

// Closures
doSomething(firstValue: 3, secondValue: 7) { $0 + $1 }

// Nested calls
myObject.theArray().insertObject(myObject.objectToInsert(), atIndex: 0)

